OMPython/__init__.py

```diff
@@ -447,3 +462,727 @@ def getClassNames(self, className=None, recursive=False, qualified=False, sort=F
+				self.simValuesList.append(solver)
+		return self.simValuesList
+	
+	def getSimulationOptions(self):
+		dic = {'Options' : self.simNamesList, 'Value' : self.simValuesList}
+		frame = pd.DataFrame(dic)		
+		return frame
+	
+	def getOptimizationOptions(self):
+		dic = {'Options' : self.optimizeOptionsNamesList, 'Value' : self.optimizeOptionsValuesList}
+		frame = pd.DataFrame(dic)
+		return frame
+	
+	def getLinearizationOptions(self):
+		dic = {'Options' : self.linearizeOptionsNamesList, 'Value' : self.linearizeOptionsValuesList}
+		frame = pd.DataFrame(dic)
+		return frame
+
+	def simulate(self):
+		if (self.inputFlag == True):
+			self.simInput()
+			getExeFile = '{}.{}'.format(self.modelName, "exe")
+			check_exeFile_ = os.path.exists(getExeFile)
+			if(check_exeFile_):
+				cmd = getExeFile + " -csvInput=" + self.csvFile
+				
+				subprocess.call(cmd, shell = False)
+				return
+			else:
+				print "Error: application file not generated yet"
+				return None
+	
+		else:
+			getExeFile = '{}.{}'.format(self.modelName, "exe")
+			check_exeFile_ = os.path.exists(getExeFile)
+			if(check_exeFile_):
+				cmd = getExeFile
+				subprocess.call(cmd, shell = False)
+				return
+			else:
+				print "Error: application file not generated yet"
+				return None
+				
+	def getSolutions(self, varList):
+		for v in varList:
+			if v == 'time':
+				continue
+			isExist = False
+			for l in self.quantitiesList:
+				if v == l.name:
+					isExist = True
+				else:
+					continue
+			if(isExist == False):
+				print '!!! ', v, ' does not exist\n'
+				return [None]*len(varList)	
+		res_mat = '_res.mat'
+		resFile = "".join([self.modelName, res_mat])
+		check_resFile_ = os.path.exists(resFile)
+		variables = ",".join(varList)
+		vars = []
+		results = []
+
+		if(check_resFile_):
+			exp = "readSimulationResult(\"" + resFile + '",{' + variables + "})"
+			res = self.getconn.sendExpression(exp)
+			npRes = np.array(res)
+			exp2 = "closeSimulationResultFile()"
+			self.getconn.sendExpression(exp2)
+			return npRes
+		else:
+			return "Error: mat file does not exist"
+					
+	def plot_(self, x_axis, y_axis): #Not complete yet
+		plt.plot(x_axis, y_axis)
+		##        plt.title(title)
+		##        plt.grid(grid)
+		##        plt.xlabel(x_label)
+		##        plt.ylabel(y_label)
+		##        plt.savefig(savefig)
+		plt.show()
+		return
+	
+	def setContinuousValues(self, names, values):
+		try:
+			errMsgFormat = 'Error!!! Incorrect format'
+			errMsgStr = 'Error!!! value should not be string'
+			checking = self.checkAvailability(names, self.cNamesList)
+			if checking is False:
+				return None
+			for v in values:
+				if isinstance(v,int) or isinstance(v,float):
+					pass
+				elif isinstance(v,str):
+					return errMsgStr
+				else:
+					return errMsgFormat
+			variability = "variability"
+			attrValue = "continuous"
+			self.setValue(names, values, variability, attrValue, self.cNamesList, self.cValuesList)
+		except Exception as e:
+			return e
```

Comment:
swallowing all exceptions silently is typically a bad pattern, but I'll stop remarking on this now.